fissa/datahandler_framebyframe.py

```diff
@@ -18,136 +18,138 @@
 from PIL import Image, ImageSequence
 
 from . import roitools
+from . import datahandler
 
 
-def image2array(image):
-    """Open a given image file as a PIL.Image instance.
+class DataHandler(datahandler.DataHandler):
```

Comment:
Great to see this being done as a subclass. But I'd actually like to see the two data handlers put in the same file.
```python
class DataHandler():
    # existing code as in other file

class DataHandlerFrameByFrame(DataHandler):
    # code as you have in this file
```
That would be much tidier. Also, you've got this weird thing at the moment where you can do
```
import datahandler_framebyframe
x = datahandler_framebyframe.DataHandler()
y = datahandler_framebyframe.datahandler.DataHandler()
class(x) == class(y)  # Returns False and x and y have different behaviours
# Not behaviour that a user of this API would expect.
```

Also, I think it might be better coding practice to use an abstract class (`DataHandlerAbstract`) and have both `DataHandler` and `DataHandlerFrameByFrame` inherit from the abstract class instead of having `DataHandlerFrameByFrame` inherit from `DataHandler`. Because they are more like siblings than parent/child. Also then we expect custom made data handler objects to inherit from `DataHandlerAbstract`, but they don't have to inherit from `DataHandler`.

You should write `DataHandlerAbstract` to have all the methods used it `DataHandler` defined as `raise NotImplementedError()`. Any methods which are common to all data handlers can be implemented for real and then the children don't have to overwrite those methods.

Reply:
Agreed. Then we can also merge the datahandler test files.

Reply:
About an abstract class -- I will implement it (and it will have only ```NotImplementedErrors```, because all the functions are different for framebyframe!), but might not use it in the tutorial. It makes a lot more sense for most users to only edit one or two of the functions used in the standard DataHandler. I'll just make a comment for advanced users that they can define an entirely new handler from DataHandlerAbstract.